pandas/tests/tools/test_to_datetime.py

```diff
@@ -609,6 +609,88 @@ def test_to_datetime_dtarr(self, tz):
         result = to_datetime(arr)
         assert result is arr
 
+    @pytest.mark.parametrize(
+        "data, expected",
+        [
+            (
+                Timestamp("2001-10-01 12:00:01.123456789"),
+                Timestamp("2001-10-01 12:00:01.123456"),
+            ),
+            (
+                datetime(2002, 10, 1, 12, 00, 1, 123456),
+                Timestamp("2002-10-01 12:00:01.123456"),
+            ),
+            ("10/01/03 12:00:01.123456789", Timestamp("2003-10-01 12:00:01.123456789")),
+        ],
+    )
+    def test_to_datetime_preserves_resolution_when_possible(self, data, expected):
```

Comment:
I'm still confused as to why this test is necessary

Isn't the issue about `to_datetime` not skipping over objects that are already `dt.datetime`?

The subsequent test also mentions resolution, but I fail to see how it's related to the issue this PR is addressing

Reply:
thanks for explaining, I'll take a closer look later in the week

Reply:
I've had a look at that comment, but to be honest I don't see why skipping `Timestamp` or `datetime` objects would be inconsistent

I think this would be considered a bug:
```
In [24]: date_str
Out[24]: '10/01/2001 12:00:01.123456789'

In [25]: ts
Out[25]: Timestamp('2001-10-01 12:00:01.123456789')

In [26]: to_datetime([date_str, ts], format='%m/%d/%Y %H:%M:%S.%f')
Out[26]: DatetimeIndex(['2001-10-01 12:00:01.123456789', '2001-10-01 12:00:01.123456'], dtype='datetime64[ns]', freq=None)
```

I'd be in favour of:
- skipping objects that are already `Timestamp` or `datetime`
- erroring if there are `Timestamp` or `datetime` objects
but not of modifying them

Reply:
The new changes skip objects that are already `Timestamp` or `datetime.datetime` objects, producing the expected output in this case: `DatetimeIndex(['2001-10-01 12:00:01.123456789', '2001-10-01 12:00:01.123456789'], dtype='datetime64[ns]', freq=None)`